internal/daemonset/daemonset.go

```diff
@@ -32,7 +33,7 @@ const (
 //go:generate mockgen -source=daemonset.go -package=daemonset -destination=mock_daemonset.go
 
 type DaemonSetCreator interface {
-	GarbageCollect(ctx context.Context, existingDS []appsv1.DaemonSet, validKernels sets.Set[string]) ([]string, error)
+	GarbageCollect(ctx context.Context, mod *kmmv1beta1.Module, existingDS []appsv1.DaemonSet, validKernels sets.Set[string]) ([]string, error)
```

Comment:
If we also need to garbage collect moduleLoader's DS so how does it work? moduleLoader is using a `moduleLoaderData`.

Reply:
moduleLoader is just a daemonset, it does not use moduleLoaderData.  All the data needed for collecting the device-plugin and module loader are present in the Module (name, namespace, version), so that's all we need.

Reply:
This comment was based on a previous one (and misslooking this PR) - https://github.com/kubernetes-sigs/kernel-module-management/pull/334/files#r1135481519.